lib/codecs/UrlCodec.js

```diff
@@ -100,12 +103,45 @@ export class Base58DidUrlCodec {
     return decodedValue;
   }
 }
+const uuidRegex = /(urn:uuid):(.*)/g;
+export class UrnUuidCodec {
+  encode(value) {
+    // break the base58 DID URL into segments
+    const matches = Array.from(value.matchAll(uuidRegex));
+    const match = matches[0];
+
+    // encode the DID URL segments
+    const encodedUrl = (urlHeaders.has(match[1])) ? urlHeaders.get(match[1]) :
+      match[1];
+    // FIXME: Assumes all UUIDs are lowercase, which will preserve the value
+    //        but might destroy signatures. Do not compress unless the UUIDs
+    //        are all lowercase?
+    const encodedUuid = uuid.parse(match[2]);
```

Comment:
Can't we simplify any of this to just ensure the string starts with `urn:uuid:` and then drop the prefix and use the `uuid` lib to do the parsing and binary formatting?

Reply:
Simplified the code here 69a2583